### PR TITLE
DM-33314: Implement missing methods for Cassandra backend

### PR DESCRIPTION
**Pull request by andy-slac**
APDB API added few methods to support replication to/from PPDB on a previous ticket. This update adds an implementation of these new methods to Cassandra backend. Cassandra code was updated to use execution profiles, as the previous model gets deprecated. Also a continuing refactoring of Cassandra implementation in a couple of commits. It makes sense to look at the final diff, some pieces were changed more than once.

### CODE REVIEW

File: python/lsst/dax/apdb/apdbCassandra.py [hunk 1]
```diff
@@ -446,32 +278,33 @@ def makeSchema(self, drop: bool = False) -> None:
...
+            if params:
+                statement = self._prep_statement(full_query)
+            else:
+                # If there are no params then it is likely that query has a
+                # bunch of literals rendered already, no point trying to
+                # prepare it because it's not reusable.
+                statement = cassandra.query.SimpleStatement(full_query)
+            statements.append((statement, params))
+        _LOG.debug("getDiaObjects: #queries: %s", len(statements))
         # _LOG.debug("getDiaObjects: queries: %s", queries)
```
Comment (n8pease) [inline python/lsst/dax/apdb/apdbCassandra.py:299]: remove commented line?

File: python/lsst/dax/apdb/apdbCassandra.py [hunk 2]
```diff
@@ -675,11 +411,252 @@ def store(self,
...
+            else:
+                query = (
+                    f'UPDATE "{self._keyspace}"."{table_name}"'
+                    ' SET "ssObjectId" = ?, "diaObjectId" = NULL'
+                    ' WHERE "apdb_part" = ? AND "apdb_time_part" = ? AND "diaSourceId" = ?'
+                )
+                values = (ssObjectId, apdb_part, apdb_time_part, diaSourceId)
+            queries.add(self._prep_statement(query), values)
+
+        # _LOG.debug("query: %s", query)
```
Comment (n8pease) [inline python/lsst/dax/apdb/apdbCassandra.py:479]: remove?

**Comment by n8pease** [comment]
LGTM!

File: python/lsst/dax/apdb/apdbCassandraSchema.py [hunk 3]
```diff
@@ -257,22 +322,15 @@ def _tableColumns(self, table_name: ApdbTables) -> List[str]:
         _LOG.debug("clust_columns: %s", clust_columns)
         if not part_columns:
             raise ValueError(f"Table {table_name} configuration is missing partition index")
-        if not clust_columns:
-            raise ValueError(f"Table {table_name} configuration is missing primary index")
+        # if not clust_columns:
+        #     raise ValueError(f"Table {table_name} configuration is missing primary index")
```
[the remark this answers is not in the document]
Comment (andy-slac) [reply python/lsst/dax/apdb/apdbCassandraSchema.py:326]: I forgot to clean that up after testing, will remove it completely.

**Pull request merged**